code/modules/admin/permissionedit.dm

```diff
@@ -413,60 +698,70 @@ ADMIN_VERB(edit_admin_permissions, R_PERMISSIONS, "Permissions Panel", "Edit adm
-					INSERT INTO [format_table_name("admin_log")] (datetime, round_id, adminckey, adminip, operation, target, log)
-					VALUES (NOW(), :round_id, :adminckey, INET_ATON(:adminip), 'add rank', :new_rank, CONCAT('New rank added: ', :new_rank))
-				"}, list("round_id" = "[GLOB.round_id]", "adminckey" = usr.ckey, "adminip" = usr.client.address, "new_rank" = custom_rank.name))
-				if(!query_add_rank_log.warn_execute())
-					qdel(query_add_rank_log)
-					return
+				return
+			QDEL_NULL(query_add_rank)
+			var/datum/db_query/query_add_rank_log = SSdbcore.NewQuery({"
+				INSERT INTO [format_table_name("admin_log")] (datetime, round_id, adminckey, adminip, operation, target, log)
+				VALUES (NOW(), :round_id, :adminckey, INET_ATON(:adminip), '[PERMISSIONS_ACTION_RANK_ADDED]', :new_rank, CONCAT('New rank added: ', :new_rank))
+			"}, list("round_id" = "[GLOB.round_id]", "adminckey" = usr.ckey, "adminip" = usr.client.address, "new_rank" = custom_rank.name))
+			if(!query_add_rank_log.warn_execute())
 				qdel(query_add_rank_log)
-			qdel(query_rank_in_db)
+				return
+			QDEL_NULL(query_add_rank_log)
 		var/datum/db_query/query_change_rank = SSdbcore.NewQuery(
 			"UPDATE [format_table_name("admin")] SET `rank` = :new_rank WHERE ckey = :admin_ckey",
 			list("new_rank" = joined_rank, "admin_ckey" = admin_ckey)
 		)
 		if(!query_change_rank.warn_execute())
 			qdel(query_change_rank)
 			return
-		qdel(query_change_rank)
+		QDEL_NULL(query_change_rank)
+		message_admins(m1)
+		log_admin(m2)
 		var/datum/db_query/query_change_rank_log = SSdbcore.NewQuery({"
 			INSERT INTO [format_table_name("admin_log")] (datetime, round_id, adminckey, adminip, operation, target, log)
-			VALUES (NOW(), :round_id, :adminckey, INET_ATON(:adminip), 'change admin rank', :target, CONCAT('Rank of ', :target, ' changed from ', :old_rank, ' to ', :new_rank))
+			VALUES (NOW(), :round_id, :adminckey, INET_ATON(:adminip), '[PERMISSIONS_ACTION_ADMIN_RANK_CHANGED]', :target, CONCAT('Rank of ', :target, ' changed from ', :old_rank, ' to ', :new_rank))
 		"}, list("round_id" = "[GLOB.round_id]", "adminckey" = usr.ckey, "adminip" = usr.client.address, "target" = admin_ckey, "old_rank" = old_rank, "new_rank" = joined_rank))
 		if(!query_change_rank_log.warn_execute())
 			qdel(query_change_rank_log)
 			return
-		qdel(query_change_rank_log)
-	if(D) //they were previously an admin
-		D.disassociate() //existing admin needs to be disassociated
-		D.ranks = new_ranks //set the admin_rank as our rank
-		D.bypass_2fa = TRUE // Another admin has cleared us
+		QDEL_NULL(query_change_rank_log)
+	else
+		message_admins(m1)
+		log_admin(m2)
+	if(target_holder) //they were previously an admin
+		target_holder.disassociate() //existing admin needs to be disassociated
+		target_holder.ranks = new_ranks //set the admin_rank as our rank
+		target_holder.bypass_2fa = TRUE // Another admin has cleared us
 		var/client/C = GLOB.directory[admin_ckey]
-		D.associate(C)
+		target_holder.associate(C)
```

Comment:
turn C to `target_client` or something maybe? you were removing a lot of single-letter variables in this refactor so just pointing this one out also

Reply:
I did miss this yea